packages/filesystem/src/browser/file-resource.ts

```diff
@@ -153,14 +156,15 @@ export class FileResource implements Resource {
             return stat.value;
         } catch (e) {
             if (e instanceof TextFileOperationError && e.textFileOperationResult === TextFileOperationResult.FILE_IS_BINARY) {
-                if (await this.shouldOpenAsText('The file is either binary or uses an unsupported text encoding.')) {
+                if (await this.shouldOpenAsText(nls.localize('theia/filesystem/fileResource/binaryTitle', 'The file is either binary or uses an unsupported text encoding.'))) {
                     this.acceptTextOnly = false;
                     return this.readStream(options);
                 }
             } else if (e instanceof FileOperationError && e.fileOperationResult === FileOperationResult.FILE_TOO_LARGE) {
                 const stat = await this.fileService.resolve(this.uri, { resolveMetadata: true });
                 const maxFileSize = GENERAL_MAX_FILE_SIZE_MB * 1024 * 1024;
-                if (this.limits?.size !== maxFileSize && await this.shouldOpenAsText(`The file is too large (${BinarySize.formatSize(stat.size)}).`)) {
+                if (this.limits?.size !== maxFileSize && await this.shouldOpenAsText(nls.localize(
+                    'theia/filesystem/fileResource/largeFileTitle', 'The file is too large ({0}).', BinarySize.formatSize(stat.size)))) {
```

Comment:
```suggestion
                if (this.limits?.size !== maxFileSize && await this.shouldOpenAsText(nls.localizeByDefault(
                    'The file is not displayed in the editor because it is too large ({0}).', BinarySize.formatSize(stat.size)))) {
```

Reply:
Unfortunately, we can't use the one from the `metadata.json` because this message isn't restricted to the editor package: any file operation that tries to access a binary file or a file that's too large will get these messages. I think that that's the wrong approach, but it's the way it's currently handled, and changing it would require a non-trivial reworking of the code that loads such files. I was tempted, though :-).

Reply:
Ah, I see, then I'll resolve those comments, thanks for the explanation.